pinot-transport/src/main/java/com/linkedin/pinot/routing/RoutingTableBuilderFactory.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.routing;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.configuration.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linkedin.pinot.common.config.TableConfig;
+import com.linkedin.pinot.common.utils.CommonConstants.Helix.TableType;
+import com.linkedin.pinot.routing.builder.BalancedRandomRoutingTableBuilder;
+import com.linkedin.pinot.routing.builder.DefaultOfflineRoutingTableBuilder;
+import com.linkedin.pinot.routing.builder.DefaultRealtimeRoutingTableBuilder;
+import com.linkedin.pinot.routing.builder.RoutingTableBuilder;
+
+public class RoutingTableBuilderFactory {
+  private static final Logger LOGGER = LoggerFactory.getLogger(RoutingTableBuilderFactory.class);
+
+  private static final Set<Class<? extends RoutingTableBuilder>> _routingTableBuilders = new HashSet<>();
+  private static final Map<String, Class<? extends RoutingTableBuilder>> map;
+
+  private Configuration _configuration;
+
+  static {
+    map = new HashMap<>();
+    _routingTableBuilders.add(BalancedRandomRoutingTableBuilder.class);
+    for (Class<? extends RoutingTableBuilder> class1 : _routingTableBuilders) {
+      map.put(class1.getSimpleName().replaceAll("RoutingTableBuilder", ""), class1);
+    }
+  }
+
+  public RoutingTableBuilderFactory(Configuration configuration) {
+    this._configuration = configuration;
+  }
+
+  public RoutingTableBuilder createRoutingTableBuilder(TableConfig tableConfig) {
+    String builderName = tableConfig.getRoutingConfig().getRoutingTableBuilderName();
+    Class<? extends RoutingTableBuilder> class1 = map.get(builderName);
+    RoutingTableBuilder builder = null;
+    if (class1 != null) {
+      try {
+        builder = class1.newInstance();
```

Comment:
Does this need to be created via reflection? Can we not use switch/case on name and call the appropriate constructor?